src/base/bittorrent/session.cpp

```diff
@@ -2368,6 +2368,36 @@ void Session::setIPFilterFile(QString path)
     }
 }
 
+void Session::setBannedIPs(const QStringList &newList)
+{
+    if (newList == m_bannedIPs)
+        return; // do nothing
+    // malformed IP strings will be silently filtered out
+    QStringList filteredList;
+    foreach (const QString &ip, newList) {
+        boost::system::error_code ec;
+        libt::address::from_string(ip.toLatin1().constData(), ec); // only error code needed
+        if (!ec) {
+            filteredList << ip;
+        }
+        else {
+            Logger::instance()->addMessage(
+                        tr("%1' is not a valid IP address and was rejected while applying the list of banned addresses.")
+                        .arg(ip)
+                        , Log::INFO);
+        }
+    }
+    // store to session settings
+    m_bannedIPs = filteredList;
+    m_IPFilteringChanged = true;
+    configureDeferred();
```

Comment:
I think that building ip_filter from scratch (reparse file and add manual part ) will be less error prone and more maintainable way to do it. Performance penalty does not matter because it will be not frequent operation, I think.